### PR TITLE
Adding documentation for how to get started with Matter & iOS/tvOS/macOS

### PR DESCRIPTION
**Pull request by woody-apple**
#### Problem
Documentation doesn't exist

#### Change overview
Add it

#### Testing
Documentation changes

### CODE REVIEW

**Comment by woody-apple** [comment]
Fast tracking documentation changes.

However, please hold off merging, reviews to come, and more updates to come.

File: docs/guides/darwin.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+-   [Simulated Linux](./simulated_device_linux.md)
+-   [Telink](/examples/lighting-app/telink/README.md)
+-   [TI Platform](./ti_platform_overview.md)
+-   [TI All Clusters](/examples/all-clusters-app/cc13x2x7_26x2x7/README.md)
+-   [Tizen](/examples/lighting-app/tizen/README.md)
+
+### Release Notes & Known Issues
+
+-   Please refer to the iOS/iPadOS 15.6
+    [Release Notes](https://developer.apple.com/documentation/ios-ipados-release-notes/ios-ipados-15_6-release-notes)
```
Comment (aajain-com) [inline docs/guides/darwin.md:176]: Please also add release notes for iOS 16.0 at https://developer.apple.com/documentation/ios-ipados-release-notes/ios-ipados-16-release-notes.

Comment (woody-apple) [reply docs/guides/darwin.md:176]: Done, ty

File: docs/guides/darwin.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
+# Testing with Apple Devices
+
+### Matter is the foundation for connected things.
+
+Learn more about Matter on the [here](https://buildwithmatter.com/)
```
Comment (bzbarsky-apple) [inline docs/guides/darwin.md:5]:
```suggestion
Learn more about Matter [here](https://buildwithmatter.com/)
```

File: docs/guides/darwin.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+In order to work with iOS/iPadOS/tvOS 15.6 or greater, device types as defined
+in the Matter Device Library spec are used to determine accessory categories.
+Ensure the right device type is set for each endpoint.
+
+-   For the `all-clusters-app` as an example, this can be set in
+    `FIXED_DEVICE_TYPES` in `endpoint_config.h`
+
+-   A light on Endpoint 1 can be set in `FIXED_DEVICE_TYPES` in
+    `endpoint_config.h` for `all-clusters-app` like so:
```
Comment (bzbarsky-apple) [inline docs/guides/darwin.md:80]: Actually, you have to set FIXED_DEVICE_TYPES, FIXED_DEVICE_TYPE_OFFSETS, and FIXED_DEVICE_TYPE_LENGTHS.

In particular, the "1" in the array below is not the endpoint id; it's the device revision number.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

File: docs/guides/darwin.md [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+| Temperature Sensor | 770     | 0302 |
+| Flow Sensor        | 774     | 0306 |
+
+#### Examples of how to setup devices
+
+##### Case study 1: Configuring a development M5Stack, as a multi-device to work with iOS/iPadOS/tvOS
+
+Note: These instructions are specific to getting started with the
+(Matter-provided) `all-clusters-app` on an ESP32-based M5Stack, however can be
+generalised to work on most platforms (more listed below)
```
Comment (carol-apple) [inline docs/guides/darwin.md:151]:
```suggestion
generalised to work on most platforms ([more listed below](#guides))
```

**Pull request merged**